tensorflow/core/kernels/data/optimize_dataset_op_test.cc

```diff
@@ -0,0 +1,143 @@
+/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow/core/kernels/data/dataset_test_base.h"
+
+namespace tensorflow {
+namespace data {
+namespace {
+
+constexpr char kNodeName[] = "optimize_dataset";
+constexpr char kOpName[] = "OptimizeDataset";
+
+class OptimizeDatasetOpTest : public DatasetOpsTestBase {
+ protected:
+  // Creates a new `OptimizeDataset` op kernel.
+  Status CreateOptimizeDatasetOpKernel(
+      const DataTypeVector& output_types,
+      const std::vector<PartialTensorShape>& output_shapes,
+      const std::vector<string>& optimization_configs,
+      std::unique_ptr<OpKernel>* optimize_dataset_op_kernel) {
+    NodeDef node_def = test::function::NDef(
+        kNodeName, kOpName, {"input_dataset", "optimizations"},
+        {{"output_types", output_types},
+         {"output_shapes", output_shapes},
+         {"optimization_configs", optimization_configs}});
+    TF_RETURN_IF_ERROR(CreateOpKernel(node_def, optimize_dataset_op_kernel));
+    return Status::OK();
+  }
+
+  // Create a new `OptimizeDataset` op kernel context
+  Status CreateOptimizeDatasetContext(
+      OpKernel* const op_kernel,
+      gtl::InlinedVector<TensorValue, 4>* const inputs,
+      std::unique_ptr<OpKernelContext>* context) {
+    TF_RETURN_IF_ERROR(CheckOpKernelInput(*op_kernel, *inputs));
+    TF_RETURN_IF_ERROR(CreateOpKernelContext(op_kernel, inputs, context));
+    return Status::OK();
+  }
+};
+
+TEST_F(OptimizeDatasetOpTest, FilterFusion) {
```

Comment:
I would prefer if the test uses the "noop_elimination" optimization (as that one is more stable than the filter fusion one and I think the test will be easier to write too).